### PR TITLE
Initialize tarball download dir into proper git repo

### PR DESCRIPTION
**Pull request by ChisSoc**

### CODE REVIEW

File: installation/routines/setup_git.sh [hunk 1]
```diff
@@ -1,15 +1,88 @@
...
+
+  # We always add origin as the selected (possible) user repository
+  # and MiczFlor's repository as upstream
+  # This means for developers everything is fully set up.
+  # For users there is no difference (origin = upstream = MiczFlor)
+  # We need to get the branch with larger depth, as we do not know
+  # how many commits happened between download and git repo init
+  if [[ $GIT_USE_SSH = true ]]; then
+      git remote add origin "git@github.com:${GIT_USER}/${GIT_REPO_NAME}.git"
+      git remote add upstream "git@github.com:MiczFlor/${GIT_REPO_NAME}.git"
```
Comment (pabera) [inline installation/routines/setup_git.sh:17]: I would prefer to outsource `MiczFlor` to its own constant as well.
Maybe we could create a `00_constant.sh` where we store such matter, in addition to [this](https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3/develop/installation/includes/01_default_config.sh#L3) 🤓

File: installation/routines/setup_git.sh [hunk 2]
```diff
@@ -1,15 +1,88 @@
...
+      git remote add upstream "https://github.com/MiczFlor/${GIT_REPO_NAME}.git"
+      git fetch origin "${GIT_BRANCH}"
+  fi
+
+  # In case we get a non-develop or non-main branch, we speculatively
+  # try to get these branches, so they can be checkout out with
+  # git checkout future3/develop
+  # without the need to set up the remote tracking information
+  # However, in a user repository, these may not be present, so we suppress output in these cases
+  if [[ $GIT_BRANCH != "future3/main" ]]; then
```
Comment (pabera) [inline installation/routines/setup_git.sh:41]: Same here .. those values might be different in the future

File: installation/routines/setup_git.sh [hunk 3]
```diff
@@ -1,15 +1,88 @@
...
+    OUTPUT=$(git fetch origin "future3/main" --depth 1 2>&1)
+    if [[ $? -ne 128 ]]; then
+      echo -e "$OUTPUT"
+    fi
+  fi
+  if [[ $GIT_BRANCH != "future3/develop" ]]; then
```
Comment (pabera) [inline installation/routines/setup_git.sh:47]: ... and here

File: installation/routines/setup_git.sh [hunk 4]
```diff
@@ -1,15 +1,88 @@
...
+  git pull origin "$GIT_BRANCH" 2>&1
+  if  [[ $? -ne 0 ]]; then
+    echo "*** Git pull ***************************************"
+    echo "FAILED to pull repository into existing directory"
+    echo "Your Box will run ok, but you need to fix the"
+    echo "git state before doing any git pulls or updating"
+    echo "****************************************************"
+  else
+    # Not necessary
+    # git branch "--set-upstream-to=origin/${GIT_BRANCH}" "$GIT_BRANCH"
```
Comment (pabera) [inline installation/routines/setup_git.sh:67]: should this remain here?

File: installation/routines/setup_git.sh (hunk 2 above)
Comment (ChisSoc) [reply installation/routines/setup_git.sh]: Done.

File: installation/routines/setup_git.sh (hunk 4 above)
Comment (ChisSoc) [reply installation/routines/setup_git.sh]: Done.

**Comment by ChisSoc** [comment]
Quite a quality update :-)
Note: This has been extensively tested stand-alone but not yet with a fresh SD-Card install

**Review by pabera — APPROVED** [review]

**Pull request merged**